contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol

```diff
@@ -0,0 +1,1456 @@
+        );
+    }
+
+    /**
+     * Transfer ETH reward to caller of the ripcord function. If the ETH balance on this contract is less than required
+     * incentive quantity, then transfer contract balance instead to prevent reverts.
+     *
+     * return uint256           Amount of ETH transferred to caller
+     */
+    function _transferEtherRewardToCaller(uint256 _etherReward) internal returns(uint256) {
+        uint256 etherToTransfer = _etherReward < address(this).balance ? _etherReward : address(this).balance;
+
+        msg.sender.transfer(etherToTransfer);
+
+        return etherToTransfer;
+    }
+
+    /**
+     * Internal function returning the ShouldRebalance enum used in shouldRebalance and shouldRebalanceWithBounds external getter functions
+     *
+     * return ShouldRebalance         Enum detailing whether to rebalance, iterateRebalance, ripcord or no action
+     */
+    function _shouldRebalance(
+        int256 _currentLeverageRatio,
+        int256 _minLeverageRatio,
+        int256 _maxLeverageRatio
+    )
+        internal
+        view
+        returns(ShouldRebalance)
+    {
+        // Get absolute value of current leverage ratio
+        uint256 currentLeverageRatioAbs = _currentLeverageRatio.abs();
+
+        // If above ripcord threshold, then check if incentivized cooldown period has elapsed
+        if (currentLeverageRatioAbs >= incentive.incentivizedLeverageRatio.abs()) {
+            if (lastTradeTimestamp.add(incentive.incentivizedTwapCooldownPeriod) < block.timestamp) {
+                return ShouldRebalance.RIPCORD;
+            }
+            return ShouldRebalance.NONE;
+        }
+
+        // If TWAP, then check if the cooldown period has elapsed
+        if (twapLeverageRatio != 0) {
+            if (lastTradeTimestamp.add(execution.twapCooldownPeriod) < block.timestamp) {
+                return ShouldRebalance.ITERATE_REBALANCE;
+            }
+            return ShouldRebalance.NONE;
+        }
+
+        // If not TWAP, then check if the rebalance interval has elapsed OR current leverage is above max leverage OR current leverage is below
+        // min leverage
+        if (
+            block.timestamp.sub(lastTradeTimestamp) > methodology.rebalanceInterval
+            || currentLeverageRatioAbs > _maxLeverageRatio.abs()
+            || currentLeverageRatioAbs < _minLeverageRatio.abs()
+        ) {
+            return ShouldRebalance.REBALANCE;
+        }
+
+        // Rebalancing is given priority over reinvestment.
+        // This might lead to scenarios where this function returns `ShouldRebalance.REINVEST` in the current block
+        // and `ShouldRebalance.REBALANCE` in the next block. In such cases, the keeper system would have to replace their
+        // reinvestment transaction with a rebalance transaction.
+        if (block.timestamp.sub(lastReinvestTimestamp) > methodology.reinvestInterval) {
```

Comment:
There's a lot of time comparison in these validators and two patterns:
```
lastTimestamp.add(intervalOrPeriod) < block.timestamp
block.timestamp.sub(lastTimestamp) > intervalOrPeriod
```
Is it possible to use one pattern and execute them all with a helper method?

Reply:
Skipped.